packages/react/kibana_context/root/root_provider.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { I18nStart } from '@kbn/core-i18n-browser';
+import React, { FC } from 'react';
+
+import { KibanaEuiProvider, type KibanaEuiProviderProps } from './eui_provider';
+
+/** Props for the KibanaRootContextProvider */
+export interface KibanaRootContextProviderProps extends KibanaEuiProviderProps {
+  /** The `I18nStart` API from `CoreStart`. */
+  i18n: I18nStart;
+}
+
+/**
+ * The `KibanaRootContextProvider` provides the necessary context for Kibana's React components, including
+ * the theme and i18n context.  This context should only be used _once_, and at the _very top_ of the
```

Comment:
Maybe could even be more explicit and say something along the lines of "this context is only designed to be used once in core; do not use it in your application"

Reply:
We're going to use it in other places... would it still be ok to list those out?  I'll massage the language and let you know.

Reply:
Not critical - I just figured if the other places are tests/storybook we could maybe be more explicit and say "don't use it in your app". If there are places where it legitimately is needed in runtime code, then probably not worth changing anything.